eng/tools/generate-dev-builds/index.js

```diff
@@ -0,0 +1,69 @@
+const path = require("path");
+const util = require("util");
+const fs = require("fs");
+const writeFile = util.promisify(fs.writeFile);
+const readFile = util.promisify(fs.readFile);
+const parse = require("jju").parse;
+const semver = require("semver");
+
+const getRushPackages = async (rushPath) => {
+  const baseDir = path.dirname(rushPath);
+  const rushJson = parse(await readFile(rushPath, "utf8"));
+  const packageData = {};
+
+  for (const proj of rushJson.projects) {
+    const filePath = path.join(baseDir, proj.projectFolder, "package.json");
+    const packageJson = parse(await readFile(filePath, "utf8"));
+    packageData[packageJson.name] = {
+      src: filePath,
+      json: packageJson,
+      newVer: undefined
+    };
+  }
+
+  return packageData;
+};
+
+const commitChanges = async (rushPackages, package) => {
+  // Commit the new version to the JSON document
+  rushPackages[package].json.version = rushPackages[package].newVer;
+  try {
+    // Write out the JSON document to disk
+    await writeFile(rushPackages[package].src, JSON.stringify(rushPackages[package].json, null, 2));
+    console.info(
+      "File " + rushPackages[package].src + " created successfully with Node.js v10 fs/promises!"
+    );
+  } catch (e) {
+    console.error(e);
+  }
+};
+
+const updatePackageVersion = (rushPackages, package, buildId) => {
+  rushPackages[package].newVer = `${rushPackages[package].json.version}-dev-${buildId}`;
+};
+
+const [
+  nodePath,   /* Ex: /bin/node */
+  scriptPath, /* /repo/common/eng/tools/generate-dev-builds/index.js */
+  buildId     /* buildNo of current build */
+] = process.argv;
+if (!nodePath || !scriptPath) {
+  throw new Error("Unexpected exception: could not detect node path or script path");
+}
+
+const main = async () => {
+  const rushPackages = await getRushPackages(path.resolve(`${__dirname}/../../../rush.json`));
+
+  // Set all the new versions
+  console.log(`Updating packages with build ID ${buildId}`);
+  for (const package of Object.keys(rushPackages)) {
+    updatePackageVersion(rushPackages, package, buildId);
+  }
+
+  // Commit the package and dependency version changes to disk
```

Comment:
Now that you aren't updating dependencies I assume you can combine the updatePackageVersion and commitChanges steps together so no need to loop more then once.

Reply:
Do we think we will want this functionality back in the future? If so I'd suggest we just remove or comment out the loop that calls `updateInternalDependencyVersions`.

Or preferably, we could squash all of the commits except the last one into a single commit, and then apply Karishma's commit that removes most of the code (the part that implements `updateInternalDependencyVersions`). Then when we merge we can create a proper merge commit or do a rebase and merge so that if we do need it in the future, we can restore it from the Git commit history.

Reply:
done both

Reply:
>Do we think we will want this functionality back in the future? 

If our current strategy works out then we should not need to bring this back so I would prefer to not have it in the source or history (we can always keep it in a branch in the fork if we do need to refer back to it later).